src/transports/smart.c

```diff
@@ -266,7 +266,12 @@ static int git_smart__connect(
 	/* We now have loaded the refs. */
 	t->have_refs = 1;
 
-	first = (git_pkt_ref *)git_vector_get(&t->refs, 0);
+	pkt = (git_pkt *)git_vector_get(&t->refs, 0);
+	if (pkt && GIT_PKT_REF != pkt->type) {
```

Comment:
We usually check the other way round, with the constant on the right-hand side. But here you seem to simply be adhering to style of surrounding code, so this is okay